src/Sylius/Bundle/ThemeBundle/Twig/ThemeFilesystemLoader.php

```diff
@@ -70,6 +72,25 @@ public function getSource($name)
     /**
      * {@inheritdoc}
      */
+    public function getSourceContext($name)
+    {
+        try {
+            $path = $this->findTemplate($name);
+
+            return new \Twig_Source(file_get_contents($path), $name, $path);
+        } catch (\Exception $exception) {
+            // In Twig 2.0, getSourceContext is part of \Twig_LoaderInterface
+            if ($this->decoratedLoader instanceof \Twig_SourceContextLoaderInterface || method_exists('\\Twig_LoaderInterface', 'getSourceContext')) {
```

Comment:
In this case, no.  The intent is to purposefully check the method exists on the interface (essentially check if the Twig 2.0 interface is in use).  It could just be a single check on the decorated loader for all cases, but IMO this purposefully makes it clear that a check is being done for a feature across multiple version branches.